cmd/argo/commands/common/wait.go

```diff
@@ -69,7 +69,16 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 			continue
 		}
 		wf := event.Object
-		if wf != nil && !wf.Status.FinishedAt.IsZero() {
+
+		isWorkflowInTerminalState := func(wf *wfv1.Workflow) bool {
```

Comment:
Can `FinishedAt` ever be non-zero when the Phase isn't completed? Is that a redundant check now?

Reply:
I believe you are right and it is redundant, removed this check. Now only checking for the phase